File: front-end/phase4/src/components/services/CourseService.js
import axios from "axios";

const getRegisteredSectionsUrl = 'http://localhost:8080/api/course/get/registeredSections/';
const getCurrentSemesterCoursesUrl = 'http://localhost:8080/api/course/get/currentSemesterCourses/';
const getAddedSectionsInBasketUrl = 'http://localhost:8080/api/course/basket/get/addedSectionsInBasket/';

const getBasketStatisticsUrl = 'http://localhost:8080/api/course/basket/get/basketStatistics/';
const insertSectionToBasketUrl = 'http://localhost:8080/api/course/basket/post/sectionToBasket/';
const deleteSectionFromBasketUrl = 'http://localhost:8080/api/course/basket/delete/sectionFromBasket/';

const getRegistrationStatusUrl = 'http://localhost:8080/api/course/basket/get/registrationStatus/';
const updateRegistrationStatusUrl = 'http://localhost:8080/api/course/basket/update/registrationStatus/';

const updateSectionQuotaUrl = 'http://localhost:8080/api/course/section/update/quota/'


class CourseService{

    getRegisteredSections(id, token){
        return axios.get(getRegisteredSectionsUrl+id, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    getCurrentSemesterCourses(year, token){
        return axios.get(getCurrentSemesterCoursesUrl+year, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    getAddedSectionsInBasket(user_id, token){
        return axios.get(getAddedSectionsInBasketUrl+user_id, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    getBasketStatistics(user_id, token){
        return axios.get(getBasketStatisticsUrl+user_id, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    insertSectionToBasket(user_id, section_id, token){
        return axios.post(insertSectionToBasketUrl+user_id+"/"+section_id, {}, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    deleteSectionFromBasket(user_id, section_id, token){
        console.log("called");
        return axios.delete(deleteSectionFromBasketUrl+user_id+"/"+section_id, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    getRegistrationStatus(user_id, token){
        return axios.get(getRegistrationStatusUrl+user_id, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    updateRegistrationStatus(user_id, token, statusDto){
        console.log(statusDto);
        return axios.put(updateRegistrationStatusUrl+user_id, {
            registration_status: statusDto.registration_status,
            advisor_approval: statusDto.advisor_approval
        }, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }

    updateSectionQuota(section_id, operation, token){
        return axios.put(updateSectionQuotaUrl+section_id+"/"+operation, {}, {
            headers: {"Authorization": `Bearer ${token}`}
        })
    }
}

export default new CourseService();

File: back-end/phase4/src/main/java/com/databaseLab/phase4/service/CourseService.java
package com.databaseLab.phase4.service;

import com.databaseLab.phase4.dto.BasketStatisticsDto;
import com.databaseLab.phase4.dto.StatusDto;
import com.databaseLab.phase4.entity.Course;
import com.databaseLab.phase4.entity.RegisteredSections;
import com.databaseLab.phase4.repository.CourseRepository;
import com.databaseLab.phase4.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CourseService {

    private final CourseRepository courseRepository;
    private final UserRepository userRepository;

    @Autowired
    public CourseService(CourseRepository courseRepository, UserRepository userRepository) {
        this.courseRepository = courseRepository;
        this.userRepository = userRepository;
    }

    public List<RegisteredSections> getRegisteredSections(int user_id){
        return courseRepository.findRegisteredSections(user_id);
    }

    public List<Course> getCurrentSemesterCourses(int year){
        return courseRepository.findCurrentSemesterCourses(year);
    }

    public List<Course> getAddedSectionsInBasket(int user_id){
        int basket_id = userRepository.findBasketIdOfStudent(user_id);
        return courseRepository.findAddedSectionsInBasket(basket_id);
    }

    public void addSectionToBasket(int user_id, int section_id){
        int basket_id = userRepository.findBasketIdOfStudent(user_id);
        courseRepository.insertSectionToBasket(section_id, basket_id);
    }

    public void deleteSectionFromBasket(int user_id, int section_id){
        int basket_id = userRepository.findBasketIdOfStudent(user_id);
        courseRepository.deleteSectionFromBasket(basket_id, section_id);
    }

    public void updateSectionQuota(int section_id, String operation){
        courseRepository.updateSectionQuota(section_id, operation);
    }

    public BasketStatisticsDto getBasketStatistics(int user_id){
        int basket_id = userRepository.findBasketIdOfStudent(user_id);
        return courseRepository.findBasketStatistics(basket_id);
    }

    public StatusDto getRegistrationStatus(int user_id){
        return courseRepository.findRegistrationStatus(user_id);
    }

    public void updateRegistrationStatus(int user_id, StatusDto statusDto){
        courseRepository.updateRegistrationStatus(user_id, statusDto);
    }
}
